bin/tictactoe

```diff
@@ -1,3 +1,37 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+def set_up
+  puts "How many players?"
+  players = gets.strip.to_i
+  if players == 0
+    game = Game.new(Computer.new("X"), Computer.new("O"))
+    game.play
+  elsif players == 1
+    puts "Human plays X and goes first. Good luck!"
+    game = Game.new(Human.new("X"), Computer.new("O"))
+    game.play
+  elsif players == 2
+    puts "X goes first, O goes second. Good luck!"
+    game = Game.new(Human.new("X"), Human.new("O"))
+    game.play
+  else
+    puts "Not a valid entry."
+    set_up
+  end
+end
+
+def play
+  puts "Welcome to TicTacToe!"
+  set_up
+  puts "Would you like to play again? (Y/N)"
+  answer = gets.strip
+  if answer == "Y"
+    play
+  else
+    puts "Goodbye"
+  end
+end
+
+play
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.